### PR TITLE
Convex Subgraph

### PR DESCRIPTION
**Pull request by harsh9200**
Schema Type - Yield
Subgraph URL - https://thegraph.com/hosted-service/subgraph/harsh9200/convex

### CODE REVIEW

**Comment by this-username-is-taken** [comment]
Can you update to schema 1.2.1?

**Comment by this-username-is-taken** [comment]
Can you add a README similar to that of other subgraphs? Also put the smart contract interaction chart in the README

File: subgraphs/convex/src/modules/Withdraw.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    withdrawTransaction.timestamp = block.timestamp;
+    withdrawTransaction.blockNumber = block.number;
+
+    withdrawTransaction.save();
+  }
+
+  return withdrawTransaction;
+}
+
+export function _Withdraw(
```
Comment (this-username-is-taken) [inline subgraphs/convex/src/modules/Withdraw.ts:62]: I'd recommend naming these lowercase to differentiate from generated code. Also why underscore in the front?

File: subgraphs/convex/src/modules/Rewards.ts [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+  poolId: BigInt,
+  block: ethereum.Block,
+  vault: VaultStore
+): BigDecimal {
+  let extraRewardsRevenueUSD = constants.BIGDECIMAL_ZERO;
+  let rewardsTokensLength = vault.rewardTokens!.length;
+
+  for (
+    let rewardTokenIdx = 1;
+    rewardTokenIdx < rewardsTokensLength;
```
Comment (this-username-is-taken) [inline subgraphs/convex/src/modules/Rewards.ts:81]: what's this?

**Comment by this-username-is-taken** [comment]
Great work!

File: subgraphs/convex/README.md [hunk 3]
```diff
@@ -57,3 +57,7 @@ Existing Subgraph
 https://thegraph.com/hosted-service/subgraph/convex-community/curve-pools
 
 https://github.com/convex-community/convex-subgraph
+
+
+## Convex Contract Interaction Diagram
+![alt text](https://cdn.discordapp.com/attachments/955977464247627826/972187577505415238/Flowchart.jpg)
```
Comment (this-username-is-taken) [inline subgraphs/convex/README.md:63]: you should put this under `docs`

Comment (this-username-is-taken) [reply subgraphs/convex/README.md:63]: this is still a link to discord?

Comment (harsh9200) [reply subgraphs/convex/README.md]: Added the reference link

**Pull request merged**